libs/@guardian/source-react-components/src/accordion/Accordion.tsx

```diff
@@ -21,6 +21,19 @@ export interface AccordionProps extends Props {
 	 */
 	hideToggleLabel?: boolean;
 	children: ReactElement[];
+	/**
+	 * Partial or complete theme to override the component's colour palette.
+	 * Sanctioned colours to change have been set out by the design system team.
+	 *
+	 *  The theme colours available to change are:
+	 *
+	 *  `textLabel`<br>
+	 *  `textBody`<br>
+	 *  `textCta`<br>
+	 *  `border`<br>
+	 *  `iconFill`
```

Comment:
it feels a bit brittle to have to keep these in sync with the theme?

it's not as elegant, but we set a default for the `theme` prop in the component, storybook can provide that in the docs:

<img width="749" alt="image" src="https://github.com/guardian/csnx/assets/867233/7264e647-c360-492f-be3a-92b44d1287dd">

how do you feel about this instead? (i'm not sure, but it's auto-generated which i do like)

Reply:
We'd still need to merge whatever is passed in via the `theme` prop with the default theme to handle partial themes so in a lot of cases we'd be merging the default theme with the default theme which feels like it could be confusing? (If we didn't allow partial themes we could assume the `theme` prop is either the default theme or another complete theme and avoid merging.)

I do like that it's auto-generated though. A similar approach is to set the `theme` prop in the story's args to the default theme, but then the `theme` prop is always set so non of the stories are testing the default scenario where it's undefined.

Reply:
We wouldn't be able to do that because the theme prop has priority over the theme provider prop. So it would override any theme provided themes.

Reply:
Another option is to do something like this 
<img width="475" alt="Screenshot 2024-02-26 at 14 45 08" src="https://github.com/guardian/csnx/assets/40648315/3b82458d-011d-483b-b2bf-c231c395997e">
<img width="1029" alt="Screenshot 2024-02-26 at 14 45 29" src="https://github.com/guardian/csnx/assets/40648315/f848175b-f8c4-4e60-8fda-e491233ef8b6">

Reply:
The merged theme function is 
<img width="795" alt="Screenshot 2024-02-26 at 15 13 15" src="https://github.com/guardian/csnx/assets/40648315/2850d15d-4abc-4077-a7d7-d4661116539d">


So, If you have 

```
<ThemeProvider theme={background: 'blue'}
<Button/>
</ThemeProvider>
```

Then currently this would get through as the `themeOverrides` AKA `theme` prop are undefined. If they became the default theme by default. Then this background colour would be overwritten.

Reply:
yeah this seems too complex to pollute the intention of PR with. it would be great if this could somehow be automated but this is much better than right now, so let's merge this.